### PR TITLE
Mention object type in error messages for missing method `object.method()`.

### PR DESCRIPTION
**Pull request by bredelings**
This is motivated by the common problem where people write `T <- readTrees("data/primates_tree.nex")`.  

Since the subscript `[1]` is missing, T is not a tree and so method calls like `T.taxa()` don't work.  This results in bug reports like #239.

This patch generates the "no method" error from `SyntaxFunctionCall::evaluateContent( )` instead of from `FunctionTable::findFunction( )`.  SyntaxFunctionCall has access to the type of the base variable, so we can add that to the error message.  We also suggest using `.methods()`.

```
> T = readTrees("primates.tree")
   Attempting to read the contents of file "primates.tree"
   Successfully read file
> T.ntips()
   Error:	Variable of type 'TimeTree[]' has no method called 'ntips'.  You can use '.methods()' to find available methods.
```

This will hopefully prevent bug reports like #239, as well as making `.methods()` more discoverable.

### CODE REVIEW

**Comment by bredelings** [comment]
Fixes #239

File: src/revlanguage/workspace/FunctionTable.cpp [hunk 1]
```diff
@@ -305,7 +305,7 @@ const Function& FunctionTable::findFunction(const std::string& name, const std::
         }
         else
         {
-            throw RbException("No function named '"+ name + "'");
+            return nullptr;
```
Comment (hoehna) [inline src/revlanguage/workspace/FunctionTable.cpp:308]: Why does this now return a nullptr instead of the error message?

Comment (bredelings) [reply src/revlanguage/workspace/FunctionTable.cpp:310]: I added the following comment:
```
            // Here we just return a nullptr to indicate failure.
            // The caller can produce better error messages, so it is the caller's job to throw an exception.
```

**Review by hoehna — APPROVED** [review]
Great, looks good.

**Pull request merged**